src/builder/webpack/config.browser.js

```diff
@@ -1,21 +1,36 @@
 // @flow
 
 import { join } from "path"
-
+import findCacheDir from "find-cache-dir"
 import commonWebpackConfig from "./config.common.js"
 import { offlinePlugin, offlineEntry } from "../../_utils/offline/webpack.js"
+import HardSourceWebpackPlugin from "hard-source-webpack-plugin"
 
 const chunkNameBrowser = "phenomic.browser"
 
 export default (config: PhenomicConfig): WebpackConfig => {
+  // Use different cache folder for hard-source-webpack-plugin on each env
+  const env = (config.production) ? "prod" : "dev"
```

Comment:
You should use a hash on the entire config object. Other flag might change and be used too like "config.static" or some others (user custom flags?). We need to be sure that we will have one cache per POSSIBLE config :)